File: UVa/UVa591.py
import sys


n = int(sys.stdin.readline().strip())
case = 1
while n != 0:
    heights = list(map(int, sys.stdin.readline().split()))
    total = sum(heights)
    avg = total // len(heights)
    movs = 0
    for H in heights:
        movs += abs(H - avg)
    movs //= 2
    print('Set #', case, sep='')
    print('The minimum number of moves is ', movs, '.', sep='', end='\n\n')
    case += 1
    n = int(sys.stdin.readline().strip())
    

File: UVa/UVa591.cpp
#include <iostream>

using namespace std;
constexpr int MAXN = 55;
int heights[MAXN] {};

inline int abs( int x ){
  return (x >= 0 ? x : -x);
}

int main(){
  ios_base::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);
  int n {}, CASE {1};
  while( cin >> n, n ){
    int total {}, movs {};
    for( int i = 0; i < n; ++ i ){
      cin >> heights[i];
      total += heights[i];
    }
    total /= n;
    for( int i = 0; i < n; ++ i )
      movs += abs(heights[i] - total);
    movs /= 2;
    cout << "Set #" << CASE ++ << '\n';
    cout << "The minimum number of moves is " << movs << ".\n\n";
  }
  return 0;
}
